whelktool/scripts/cleanups/2020/12/lxl-2612-byggnadsstyrelsen-broken-links.groovy

```diff
@@ -0,0 +1,31 @@
+import whelk.util.DocumentUtil
+
+String where = """
+    collection = 'bib' 
+    AND (data#>>'{@graph,1,associatedMedia}' LIKE '%http://www.fastighetswiki.se/wiki%' 
+        OR data#>>'{@graph,1,marc:versionOfResource}' LIKE '%http://www.fastighetswiki.se/wiki%'
+        OR data#>>'{@graph,1,isPrimaryTopicOf}' LIKE '%http://www.fastighetswiki.se/wiki%'
+        OR data#>>'{@graph,1,relatedTo}' LIKE '%http://www.fastighetswiki.se/wiki%')
+    """
+
+List keys856 = ['associatedMedia', 'marc:versionOfResource', 'isPrimaryTopicOf', 'relatedTo']
+
+selectBySqlWhere(where) { data ->
+    Map instance = data.graph[1]
+    Map _856 = instance.findAll { it.getKey() in keys856 }
+
+    DocumentUtil.traverse(_856) { value, path ->
+        if (value instanceof String && value.startsWith('http')) {
+            //All URLs start with 'http://www.fastighetswiki.se/wiki', thus there are no other working links via 856
+            assert value.startsWith('http://www.fastighetswiki.se/wiki')
+
+            if (instance['@type'] == 'Electronic') {
+                data.scheduleDelete()
+            } else {
+                instance.remove(path.find { it in keys856 })
+                instance.remove('otherPhysicalFormat')
```

Comment:
Are we sure that otherPhysicalFormat always refers to the fastighetswiki record?

Reply:
@olovy good point! I did check (but didn't include in the script) that otherPhysicalFormat has only one object in each of these instances. Can it be assumed then that this one object represents the fastighetswiki record? If not, how can we check this?

Reply:
It should be possible to check describedBy - Item - recordControlNumber

The weird thing about the example record in the issue is that it refers to itself.
https://libris.kb.se/katalogisering/j2vwcs7v2b3mrfr

Reply:
@olovy changed the script so that it now verifies that otherPhysicalFormat refers to a fastighetswiki record, whether electronic or physical (some do refer to their physical selves which is a bit weird, yes). A bit ad hoc now but should at least catch all cases.

Reply:
:thumbs